p2p/transport/websocket/websocket_test.go

```diff
@@ -548,3 +549,101 @@ func TestResolveMultiaddr(t *testing.T) {
+	require.NoError(t, err)
+	c := make(chan int, noOfClientConns)
+	raddr, listeners, err := startListeners(t, tpt)
+	require.NoErrorf(t, err, "failed to start listeners")
+	for i := 0; i < 2; i++ {
+		go func(index int, ln *manet.Listener, ch chan int) {
+			l := *ln
+			defer l.Close()
+			/* Looping noOfClientConns times to ensure all 4 connections are handled.
+				With SO_REUSEPORT the distribution happens based on threads that are waiting on Accept call as mentioned below.
+				We cannot gaurantee when Server go-routines would block on Accept.
+				Sometimes the client routines get scheduled first causing unequal distribution of connections.
+				Ref: https://lwn.net/Articles/542629/
+				By contrast, the SO_REUSEPORT implementation distributes connections evenly across all of the threads (or processes)
+			 	that are blocked in accept() on the same port. */
+			for j := 0; j < noOfClientConns; j++ {
+				//j := 0
+				conn, err := l.Accept()
+				require.NoErrorf(t, err, "Server Routine-", index, " Failed accepting connection ", j, " due to error ")
+				defer conn.Close()
+				buf := make([]byte, 6)
+				n, err := conn.Read(buf)
+				require.NoError(t, err)
+				require.Equal(t, 6, n)
+				n, err = conn.Write(buf)
+				require.NoError(t, err)
+				require.Equal(t, 6, n)
+				ch <- index
+			}
+
+		}(i, listeners[i], c)
+	}
+
+	for i := 0; i < noOfClientConns; i++ {
+		go func(index int) {
+			conn, err := tpt.maDial(context.Background(), raddr)
+			require.NoError(t, err)
+			defer conn.Close()
+			msg := fmt.Sprintf("Hello%d", index)
+			n, err := conn.Write([]byte(msg))
+			require.Equal(t, 6, n)
+			require.NoError(t, err)
+			buf := make([]byte, 6)
+			n, err = conn.Read(buf)
+			require.NoError(t, err)
+			require.Equal(t, 6, n)
+		}(i)
+	}
+	var connsHandled [2]int
+	//Waiting to ensure all 4 connections are handled.
+	for i := 0; i < noOfClientConns; i++ {
+		temp := <-c
+		connsHandled[temp]++
+	}
+	/*
+			 For windows and macOS load balancing is not done by kernel as per references below.
+			 For other architectures, behaviour is not known.
+			 For ubuntu load balancing doesn't seem to happen consistently.
+			 In order to not have a flaky test, commenting this additiona check.
+			 Hence, Check for load balancing only for linux based architectures.
+			 Refer https://learn.microsoft.com/en-us/windows/win32/winsock/using-so-reuseaddr-and-so-exclusiveaddruse?redirectedfrom=MSDN for windows
+			 References for MACOS
+				Ref - (https://stackoverflow.com/questions/41247790/so-reuseport-on-macos-with-libuv)
+				Ref -(https://github.com/uNetworking/uWebSockets/issues/1194)
+		if runtime.GOOS == "linux" {
+			for i := 0; i < 2; i++ {
+				/*Not checking for equal distribution of connections due to above explanation.*/
+	/*require.NotEqualf(t, 0, connsHandled[i], "No connections handled by listener %d", i)
+			t.Logf("Listener %d handled %d connections.", i, connsHandled[i])
+		}
```

Comment:
This test is fine. The distribution will never be exactly equal, the allocation is based on the hash of `clientIP:Port` so it's not round robin. You can make 20 connections and assert both the listeners get at least 1.

Reply:
On this test, can we use a separate transport for dialing? Calling dial to connect to a listener on the same transport is uncommon.

Reply:
> This test is fine. The distribution will never be exactly equal, the allocation is based on the hash of `clientIP:Port` so it's not round robin. You can make 20 connections and assert both the listeners get at least 1.

That is what i initially thought, but if you look at the CI tests no distribution is happening sometimes which is making the test flaky. Hence i had commented it out to avoid failure of the test in linux. e.g Refer here https://github.com/libp2p/go-libp2p/actions/runs/5885440443/job/15969152974

Reply:
I think that's because all the ports selected for dialing are hashing to the same slot. If we increase the number of ports(clients in this case) this should go away.